subsys/pm/device_runtime.c

```diff
@@ -127,71 +128,93 @@ static int pm_device_request(const struct device *dev,
 						  PM_DEVICE_STATE_SUSPEND,
 						  NULL, NULL);
 		}
-		ret = 0;
 		goto out;
 	}
 
-	(void)k_work_schedule(&dev->pm->work, K_NO_WAIT);
+	(void)k_mutex_lock(&dev->pm->lock, K_FOREVER);
+
+	if (!dev->pm->enable) {
```

Comment:
why does this need to be called with the lock? We could just move the lock below this block.

Reply:
nope, if a different thread disables runtime pm ?